### PR TITLE
#178. Kotlinify ReportActivity.

### PR DESCRIPTION
**Pull request by bogdan-evtushenko**

### CODE REVIEW

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+class ReportActivity : BaseBackActivity() {
+
+    @Inject
+    lateinit var recordController: RecordController
+    @Inject
+    lateinit var rateController: ExchangeRateController
+    @Inject
+    lateinit var currencyController: CurrencyController
+
+    private lateinit var recordList: List<Record>
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt]: I would make `recordList` not `lateinit`, but just assign an empty `listOf`. And `period` is actually `lateinit`.

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    private var period: Period? = null
+
+    private lateinit var shortSummaryPresenter: ShortSummaryPresenter
+
+    override fun getContentViewId(): Int {
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt]: Can be simplified to `override fun getContentViewId() = R.layout.activity_report`.

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+        return true
+    }
+
+    override fun initViews() {
+        super.initViews()
+
+        initSpinnerCurrency()
+
+        shortSummaryPresenter = ShortSummaryPresenter(this)
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt]: Probably can be moved to the place, where it's declared.

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        expListView.addHeaderView(shortSummaryPresenter.create(false, null))
+    }
+
+    private fun update(currency: String) {
+        val reportMaker = ReportMaker(rateController)
+        val report = reportMaker.getRecordReport(currency, period, recordList)
+
+        var adapter: ExpandableListReportAdapter? = null
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt]: All code below can be moved into `if` statement.

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        }
+
+        expListView.setAdapter(adapter)
+        shortSummaryPresenter.update(report, currency, reportMaker.currencyNeeded(currency, recordList))
+    }
+
+    private fun initSpinnerCurrency() {
+        val currencyList = currencyController.readAll()
+
+        spinnerCurrency.adapter = ArrayAdapter(this,
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt]: Strange line-break.

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                R.layout.view_spinner_item, currencyList)
+        spinnerCurrency.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(p0: AdapterView<*>?, p1: View?, p2: Int, p3: Long) {
+                update(spinnerCurrency.selectedItem.toString())
+            }
+
+            override fun onNothingSelected(p0: AdapterView<*>?) {
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt]: Can be one-lined.

File: app/src/main/res/layout/activity_report.xml [hunk 7]
```diff
@@ -19,7 +19,7 @@
             app:popupTheme="@style/AppTheme.PopupOverlay">
 
             <android.support.v7.widget.AppCompatSpinner
-                android:id="@+id/spinner_currency"
+                android:id="@+id/spinnerCurrency"
```
Comment (yevhenii-kanivets) [inline app/src/main/res/layout/activity_report.xml]: 👍

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt (hunk 1 above)
Comment (bogdan-evtushenko) [reply app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt]: Period need to be optional

Comment (yevhenii-kanivets) [reply app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/ReportActivity.kt]: OK

**Pull request merged**